runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java

```diff
@@ -299,6 +299,67 @@ public Response submitEvent(@PathParam("machineId") String machineId,
         return Response.status(Response.Status.ACCEPTED).build();
     }
 
+    /**
+     * Update EventData of the specified Event name under the specified State machine
+     *
+     * @param machineId the state machine identifier
+     * @return Response with execution status code
+     * @throws Exception
+     */
+    @POST
+    @Path("/{machineId}/context/eventupdate")
+    @Timed
+    public Response updateEvent(@PathParam("machineId") String machineId,
+                                EventData eventData
+    ) throws Exception {
+        try {
+            LoggingUtils.registerStateMachineIdForLogging(machineId);
+            if (eventData.getData() == null || eventData.getName() == null || eventData.getEventSource() == null) {
+                return Response.status(Response.Status.BAD_REQUEST.getStatusCode()).entity(
+                        "Event Data|Name|Source cannot be null.").build();
+            }
+            if (!(eventsDAO.findBySMIdAndName(machineId, eventData.getName()).getStatus() == Event.EventStatus.triggered)) {
+                return Response.status(Response.Status.BAD_REQUEST.getStatusCode()).entity(
+                        "Input event is not in triggered state.").build();
+            }
+            boolean canUpdateEventData = false;
+            /* List holds objects retrieved from States matching input machineId and eventName as [taskId, machineId, status] */
+            List<Object[]> states = parallelScatterGatherQueryHelper.findStatesByDependentEvent(machineId, eventData.getName());
```

Comment:
Why are Generic not used? Why objects?

Reply:
Can be done as part of refactoring exercise, it's used everywhere as Object now.